ops/pebble.py

```diff
@@ -808,7 +790,7 @@ class Layer:
     log_targets: Dict[str, 'LogTarget']
 
     def __init__(self, raw: Optional[Union[str, 'LayerDict']] = None):
-        if isinstance(raw, str):
+        if isinstance(raw, str):  # noqa: SIM108
```

Comment:
I see that this is from flake8-simplify. I think we should probably exclude that additional linters in this PR, and discuss in a separate PR.

Perhaps that's a good general comment. Would be good to separate out "turning on additional checks" (even if they're good ones) from "switching to Ruff". I think we should *mostly* restrict this PR to the latter.

Reply:
I agree, I guess I just have a slightly smaller "mostly" than you :laughing:. I'll split this into two, a very minimal one first, and then the ones I think are simple and worth doing immediately, and still leave the ones I think need more discussion to later.